### ISSUE TITLE
Help with the correct parameters for X forwarding

### ISSUE THREAD

**Issue by chernesk** [mslocrian/docker-openconnect-vpn#4]
Here's what I'm seeing:

```
vpn  | 
vpn  | (gp-saml-gui:7): dbind-WARNING **: 12:31:02.268: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-nFH7zazPon: No such file or directory
vpn  | Traceback (most recent call last):
vpn  |   File "/usr/local/lib/python3.9/dist-packages/urllib3/connection.py", line 174, in _new_conn
vpn  |     conn = connection.create_connection(
vpn  |   File "/usr/local/lib/python3.9/dist-packages/urllib3/util/connection.py", line 72, in create_connection
vpn  |     for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
vpn  |   File "/usr/lib/python3.9/socket.py", line 953, in getaddrinfo
vpn  |     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
vpn  | socket.gaierror: [Errno -2] Name or service not known
vpn  | 
vpn  | During handling of the above exception, another exception occurred:
vpn  | 
vpn  | Traceback (most recent call last):
vpn  |   File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 703, in urlopen
vpn  |     httplib_response = self._make_request(
vpn  |   File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 386, in _make_request
vpn  |     self._validate_conn(conn)
vpn  |   File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 1040, in _validate_conn
vpn  |     conn.connect()
vpn  |   File "/usr/local/lib/python3.9/dist-packages/urllib3/connection.py", line 358, in connect
vpn  |     self.sock = conn = self._new_conn()
vpn  |   File "/usr/local/lib/python3.9/dist-packages/urllib3/connection.py", line 186, in _new_conn
vpn  |     raise NewConnectionError(
vpn  | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7ff235555880>: Failed to establish a new connection: [Errno -2] Name or service not known
vpn  | 
vpn  | During handling of the above exception, another exception occurred:
vpn  | 
vpn  | Traceback (most recent call last):
vpn  |   File "/usr/local/lib/python3.9/dist-packages/requests/adapters.py", line 440, in send
vpn  |     resp = conn.urlopen(
vpn  |   File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 785, in urlopen
vpn  |     retries = retries.increment(
vpn  |   File "/usr/local/lib/python3.9/dist-packages/urllib3/util/retry.py", line 592, in increment
vpn  |     raise MaxRetryError(_pool, url, error or ResponseError(cause))
vpn  | urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='%22vpn.myschoolnamehere.edu%22', port=443): Max retries exceeded with url: /ssl-vpn/prelogin.esp (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7ff235555880>: Failed to establish a new connection: [Errno -2] Name or service not known'))
vpn  | 
vpn  | During handling of the above exception, another exception occurred:
vpn  | 
vpn  | Traceback (most recent call last):
vpn  |   File "/usr/local/bin/gp-saml-gui", line 8, in <module>
vpn  |     sys.exit(main())
vpn  |   File "/usr/local/lib/python3.9/dist-packages/gp_saml_gui.py", line 256, in main
vpn  |     res = s.post(endpoint, verify=args.verify, data=data)
vpn  |   File "/usr/local/lib/python3.9/dist-packages/requests/sessions.py", line 577, in post
vpn  |     return self.request('POST', url, data=data, json=json, **kwargs)
vpn  |   File "/usr/local/lib/python3.9/dist-packages/requests/sessions.py", line 529, in request
vpn  |     resp = self.send(prep, **send_kwargs)
vpn  |   File "/usr/local/lib/python3.9/dist-packages/requests/sessions.py", line 645, in send
vpn  |     r = adapter.send(request, **kwargs)
vpn  |   File "/usr/local/lib/python3.9/dist-packages/requests/adapters.py", line 519, in send
vpn  |     raise ConnectionError(e, request=request)
vpn  | requests.exceptions.ConnectionError: HTTPSConnectionPool(host='%22vpn.myschoolnamehere.edu%22', port=443): Max retries exceeded with url: /ssl-vpn/prelogin.esp (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7ff235555880>: Failed to establish a new connection: [Errno -2] Name or service not known'))
vpn  | Could not obtain SAML token!

```
I'm trying to discern whether I'm dealing with a QT/X issue or a nameserver issue or both. Obviously, I didn't want to put the name of the school that I'm trying to connect to in an issue, but I do have FQDN vpn host correct.

Thanks for your help!

**Comment by mslocrian** [comment]
Hi @chernesk,

This *appears* to be DNS related inside the container, especially considering the tracebacks we're seeing. 

`requests.exceptions.ConnectionError: HTTPSConnectionPool(host='%22vpn.myschoolnamehere.edu%22', port=443): Max retries exceeded with url: /ssl-vpn/prelogin.esp (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7ff235555880>: Failed to establish a new connection: [Errno -2] Name or service not known'))`

Are you able to jump into the container and see if DNS is functioning properly?

thanks,
stegen

**Comment by chernesk** [comment]
Unfortunately, no. It exits.

`vpn exited with code 1`

I was afraid of this. I'll try to figure out what might be going on. If you have any debugging tips that don't require a running container, that would be great. Otherwise, I'll see what I can find out.

I really appreciate the quick response!

**Comment by chernesk** [comment]
I think that I'm going to reboot my machine to take that out of the equation as well. :smiley:

**Comment by mslocrian** [comment]
Try doing something like this:
`% docker run mslocrian/docker-openconnect-vpn /bin/bash -c 'sleep 1000'`

Then something like:
```
% docker ps       
CONTAINER ID   IMAGE                              COMMAND                  CREATED          STATUS                             PORTS     NAMES
dd97256d2db3   mslocrian/docker-openconnect-vpn   "/bin/bash -c 'sleep…"   12 seconds ago   Up 10 seconds (health: starting)             condescending_galois
% docker exec -ti condescending_galois bash
root@dd97256d2db3:/vpn# ping google.com
PING google.com (142.250.217.110) 56(84) bytes of data.
64 bytes from sea09s30-in-f14.1e100.net (142.250.217.110): icmp_seq=1 ttl=56 time=13.5 ms
64 bytes from sea09s30-in-f14.1e100.net (142.250.217.110): icmp_seq=2 ttl=56 time=17.5 ms
64 bytes from sea09s30-in-f14.1e100.net (142.250.217.110): icmp_seq=3 ttl=56 time=13.5 ms
^C64 bytes from 142.250.217.110: icmp_seq=4 ttl=56 time=13.9 ms

--- google.com ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 15111ms
rtt min/avg/max/mdev = 13.490/14.600/17.498/1.681 ms
root@dd97256d2db3:/vpn# 
```

**Comment by chernesk** [comment]
Thanks. I learned something new today. I didn't know I could do that inside of a compose file as well.

Looking at it later, I didn't realize that quotes around the hostname was causing the problem. I finally saw the `%22hostname%22` and that got me to `gp-saml-gui: error: Gateway interface does not exist; specify --portal instead` working through probably some bad config there too.

**Comment by chernesk** [comment]
@mslocrian Thank you so much for your help before. 

I've picked this up again and am trying again to see if I can get this to work. I've not been able to do so yet.

Right now, everything seems to go decently when I start up:
```
Attaching to vpn
vpn  | + gp-saml-gui --portal --clientos=Windows vpn.myhostname.com
vpn  | 
vpn  | (gp-saml-gui:7): dbind-WARNING **: 11:08:49.497: Couldn't connect to accessibility bus: Failed to connect to socket /run/user/1000/at-spi/bus_0: No such file or directory
vpn  | QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
```
but I get a blank screen coming from X"

![image](https://user-images.githubusercontent.com/296723/181343205-f2aa1910-a51b-4ec0-9cfe-caaa3831ee52.png)

Have you experienced this at all? I really appreciate you pointing me in the right direction before and I appreciate the time that you took to help out on a simple issue. I'm sure that I'm just missing a config option here as well. 

Is there a better way that you know of to debug this?

When I'm using GlobalProtect on the desktop, the popup comes up just fine:

![image](https://user-images.githubusercontent.com/296723/181343530-1ced7d7f-bed3-40aa-ae71-69d213e0e602.png)

Sorry about the missing rest of the Window. I'm not sure how much my client would like their info on the internet. I'm really looking forward to getting this working and getting the GP client uninstalled from my desktop. ;-)

Thanks again,
  -Chris

```
